### PR TITLE
[MODULES-270] fixing problem where ClassLoader.definePackage() can th…

### PR DESCRIPTION
**Pull request by ropalka**
…row IllegalArgumentException on JDK9

### CODE REVIEW

File: src/main/java/org/jboss/modules/ConcurrentClassLoader.java [hunk 1]
```diff
@@ -455,7 +455,12 @@ protected Package definePackage(final String name, final String specTitle, final
             if (existing != null) {
                 return existing;
             }
-            Package pkg = super.definePackage(name, specTitle, specVersion, specVendor, implTitle, implVersion, implVendor, sealBase);
+            Package pkg;
+            try {
+                pkg = super.definePackage(name, specTitle, specVersion, specVendor, implTitle, implVersion, implVendor, sealBase);
+            } catch (final IllegalArgumentException iae) {
+                pkg = super.getPackage(name);
+            }
```
Comment (dmlloyd) [inline src/main/java/org/jboss/modules/ConcurrentClassLoader.java]: When does this occur?  Is this something new in Java 9?

Comment (ropalka) [inline src/main/java/org/jboss/modules/ConcurrentClassLoader.java]: I noticed that on JDK8 whole method ClassLoader.definePackage() was synchronized.
In JDK9 they're using ConcurrentHashMap.putIfAbsent() instead so it opens a window
for potential concurrent update of the underlying package map. I think this design change is the reason why we can see IllegalArgumentException in some tests.

Comment (dmlloyd) [inline src/main/java/org/jboss/modules/ConcurrentClassLoader.java]: Is there any chance an IllegalArgumentException can be thrown without the package being defined in the superclass?

Comment (dmlloyd) [inline src/main/java/org/jboss/modules/ConcurrentClassLoader.java]: I think we might want to add an ``if (pkg == null) throw iae;" after that getPackage call.

Comment (ropalka) [inline src/main/java/org/jboss/modules/ConcurrentClassLoader.java]: Sure, reworking ...

Comment (ropalka) [inline src/main/java/org/jboss/modules/ConcurrentClassLoader.java]: Done.

**Pull request merged**